### DISCUSSION TITLE
1.9inch_ESP32-1732S019 NOT WORK

### DISCUSSION THREAD

**Discussion by taoloo897** [Bodmer/TFT_eSPI#2939] [New processor/display support or capability request]
HI. I tried to use T_eSPI with 1.9inch_ESP32-1732S019 with ST7789 but doesn't work. I use this library with T-HMI ESP32-S3 Touch Display 2.8 inch ST7789 and it works GREATE.
I make it works with Arduino_GFX_Library.h but it is hard to work with it in the code.
This is the board
https://www.aliexpress.com/item/1005005951347460.html?spm=a2g0o.order_list.order_list_main.13.21ef1802v1yiy9
If you can add it in your library will be awsome.
Thanks in advance.

**Comment by Bodmer** [comment]
The display should work.

Post your setup file.

Run Read_User_Setup sketch and post the serial monitor output here.

**Comment by taoloo897** [comment]
Hi and thanks for your answer.
I have a big problem with serial monitor. I tried a lot of examples before wrote here and nou tried Reqad_User_Setup , but it is very strange , i don;t receive anything on serial only 
18:11:49.230 -> ESP-ROM:esp32s3-20210327
18:11:49.230 -> Build:Mar 27 2021
18:11:49.230 -> rst:0x1 (POWERON),boot:0xb (SPI_FAST_FLASH_BOOT)
18:11:49.230 -> SPIWP:0xee
18:11:49.230 -> mode:DIO, clock div:1
18:11:49.230 -> load:0x3fce3808,len:0x44c
18:11:49.230 -> load:0x403c9700,len:0xbd8
18:11:49.230 -> load:0x403cc700,len:0x2a80
18:11:49.230 -> entry 0x403c98d0
The baudrate is ok and i use configuration of board like in attached picture.
![Untitled](https://github.com/Bodmer/TFT_eSPI/assets/12483880/61a07600-082e-4ad2-a637-2b17c331b2be)

All what i upload from Arduino IDE not send message on serial monitor. Only if i upload the bin from vendor example with flash_download_tool_3.9.3 , than it send on serial.
Thanks  again for your time.

**Reply by Bodmer** [reply]
Did you set the baud rate correctly to 115200?

![image](https://github.com/Bodmer/TFT_eSPI/assets/15824805/5bd8a276-76fe-45f2-b687-e6876abf0d71)

**Comment by taoloo897** [comment]
Yes the baud rate is correct setted. i tried all baud from 9660 to 500000. It was from this enabled settings USB CDC on boot (strange because for lilygo esp32-s3 t-hmi 2.8 must enabled it) . I disabled it for this bord. I got the serial, but looks strange too , because is i run the same code with for lilygo esp32-s3 t-hmi 2.8 they show the same things on the serial like next:
08:30:22.741 -> [code]
08:30:22.741 -> TFT_eSPI ver = 2.5.23
08:30:22.741 -> Processor    = ESP32
08:30:22.741 -> Frequency    = 240MHz
08:30:22.741 -> Transactions = Yes
08:30:22.741 -> Interface    = Parallel
08:30:22.741 -> Display driver = 7789
08:30:22.741 -> Display width  = 240
08:30:22.741 -> Display height = 320
08:30:22.741 -> 
08:30:22.741 -> TFT_CS   = GPIO 6
08:30:22.741 -> TFT_DC   = GPIO 7
08:30:22.741 -> TFT_WR   = GPIO 8
08:30:22.741 -> 
08:30:22.741 -> TFT_D0   = GPIO 48
08:30:22.741 -> TFT_D1   = GPIO 47
08:30:22.741 -> TFT_D2   = GPIO 39
08:30:22.741 -> TFT_D3   = GPIO 40
08:30:22.741 -> TFT_D4   = GPIO 41
08:30:22.741 -> TFT_D5   = GPIO 42
08:30:22.776 -> TFT_D6   = GPIO 45
08:30:22.776 -> TFT_D7   = GPIO 46
08:30:22.776 -> 
08:30:22.776 -> TFT_BL           = GPIO 38
08:30:22.776 -> 
08:30:22.776 -> Font GLCD   loaded
08:30:22.776 -> Font 2      loaded
08:30:22.776 -> Font 4      loaded
08:30:22.776 -> Font 6      loaded
08:30:22.776 -> Font 7      loaded
08:30:22.776 -> Font 8      loaded
08:30:22.776 -> Smooth font enabled

**Reply by Bodmer** [reply]
The library is not configured correctly for the SPI display.

Edit User_Setup_Select.h to point to ***one*** suitable setup file, maybe setup 206. You will need to adjust the pins used in that setup file as it is for a different board.

**Comment by taoloo897** [comment]
With 206 setup i receive these:
4:51:45.687 -> [code]
14:51:45.687 -> TFT_eSPI ver = 2.5.23
14:51:45.687 -> Processor    = ESP32
14:51:45.687 -> Frequency    = 240MHz
14:51:45.687 -> Transactions = Yes
14:51:45.687 -> Interface    = Parallel
14:51:45.687 -> Display driver = 7789
14:51:45.687 -> Display width  = 170
14:51:45.687 -> Display height = 320
14:51:45.687 -> 
14:51:45.687 -> R0 x offset = 35
14:51:45.719 -> TFT_DC   = GPIO 7
14:51:45.719 -> TFT_RST  = GPIO 5
14:51:45.719 -> TFT_WR   = GPIO 8
14:51:45.719 -> TFT_RD   = GPIO 9
14:51:45.719 -> 
14:51:45.719 -> TFT_D0   = GPIO 39
14:51:45.719 -> TFT_D1   = GPIO 40
14:51:45.719 -> TFT_D2   = GPIO 41
14:51:45.719 -> TFT_D3   = GPIO 42
14:51:45.719 -> TFT_D4   = GPIO 45
14:51:45.719 -> TFT_D5   = GPIO 46
14:51:45.719 -> TFT_D6   = GPIO 47
14:51:45.719 -> TFT_D7   = GPIO 48
14:51:45.719 -> 
14:51:45.719 -> TFT_BL           = GPIO 38
14:51:45.719 -> TFT_BACKLIGHT_ON = HIGH
14:51:45.719 -> 
14:51:45.719 -> Font GLCD   loaded
14:51:45.719 -> Font 2      loaded
14:51:45.719 -> Font 4      loaded
14:51:45.752 -> Font 6      loaded
14:51:45.752 -> Font 7      loaded
14:51:45.752 -> Font 8      loaded
14:51:45.752 -> Smooth font enabled
But if i tried any example from tft_eSpi library , nothing show on display.
For lilygo esp32-s3 t-hmi 2.8 " every time i must put in setup next lines because without them the display not open. Maybe i must have something like this for 1.9" board but what pins??
pinMode(10, OUTPUT);
digitalWrite(10, HIGH);

**Comment by taoloo897** [comment]
Try again please.
I made it work but if i use button with some attachclick it reboot if i try to draw strings or numbers in some places, but maybe it is from my code.
[Setup72_ESP32_ST7789_172x320.zip](https://github.com/Bodmer/TFT_eSPI/files/13230484/Setup72_ESP32_ST7789_172x320.zip)

**Reply by Bodmer** [reply]
Yes, that is good. Thanks.

**Comment by Consoli93** [comment]
Could you provide me with the files? I can't do it
esp32-1732S019

**Comment by sardor1122** [comment]
Hello everyone, I was able to make it work, here is the link to the library
https://github.com/moononournation/Arduino_GFX
And this is ino file
[sketch_aug25a.zip](https://github.com/user-attachments/files/17074520/sketch_aug25a.zip)
